src/main/java/com/ibm/disni/verbs/impl/RdmaVerbsNat.java

```diff
@@ -217,4 +217,27 @@ public int destroyCQ(IbvCQ cq) throws IOException {
 		int ret = nativeDispatcher._destroyCQ(cqImpl.getObjId());
 		return ret;
 	}
+
+	@Override
+	public void checkResources(IbvContext context, int maxWR, int maxSge, int cqSize) throws IOException {
+		NatIbvContext natContext = (NatIbvContext)context;
+		int ret = nativeDispatcher._checkResources(natContext.getObjId(), maxWR, maxSge, cqSize);
+		if (ret == 0) {
+			return;
+		}
+		String warning = "";
+		if ((ret & 1) != 0)
+		    warning = "too many maxWR";
+		if ((ret & 2) != 0) {
+			if (!warning.isEmpty())
+				warning = warning + " and ";
+			warning = warning + "too many maxSge";
+		}
+		if ((ret & 4) != 0) {
+			if (!warning.isEmpty())
+				warning = warning + " and ";
+			warning = warning + "too many cqSize";
+		}
+        throw new IOException(warning);
```

Comment:
Maybe throw this from libdisni and print maxSge elements device supported.

Reply:
By default, libdisni does not print message, so users may not figure out what's going wrong. Personally I prefer throwing out an explicit exception in Java land.